lib/logstash/filters/split.rb

```diff
@@ -66,6 +66,8 @@ def register
   public
   def filter(event)
     
+    # Convert field to a string based on any format values    
+    @field = event.sprintf(@field)
```

Comment:
I tried on my side with the following example : {"array":["single-element"],"untouched":"1\n2\n3"} and it success on both tests (depending on which field I choose to split)
I can't understand why it doesn't work during the given test...